### PR TITLE
Sov 1096 notification stack

### PR DESCRIPTION
**Pull request by Rickk137**
https://sovryn.atlassian.net/browse/SOV-1096

### CODE REVIEW

File: apps/frontend/src/contexts/NotificationContext.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+const NotificationContext = createContext<NotificationContextInterface | null>(
+  null,
+);
+
+export const useNotificationContext = () => {
+  return useContext(NotificationContext) as NotificationContextInterface;
+};
+
+interface Props {
```
Comment (soulBit) [inline apps/frontend/src/contexts/NotificationContext.tsx]:
```suggestion
interface NotificationProviderProps {
```

File: apps/frontend/src/contexts/NotificationContext.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  children?: React.ReactNode;
+}
+
+export const NotificationProvider: React.FC<Props> = ({ children }) => {
```
Comment (soulBit) [inline apps/frontend/src/contexts/NotificationContext.tsx]:
```suggestion
export const NotificationProvider: React.FC<NotificationProviderProps> = ({ children }) => {
```

File: apps/frontend/src/contexts/NotificationContext.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  const [time, setTime] = useState(new Date().getTime());
```
Comment (soulBit) [inline apps/frontend/src/contexts/NotificationContext.tsx]:
```suggestion
  const [time, setTime] = useState(Date.now());
```

File: apps/frontend/src/contexts/NotificationContext.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  const [notifications, setNotifications] = useState<Notification[]>([]);
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      setTime(new Date().getTime());
```
Comment (soulBit) [inline apps/frontend/src/contexts/NotificationContext.tsx]:
```suggestion
      setTime(Date.now());
```

File: apps/frontend/src/contexts/NotificationContext.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    }, 1000);
+
+    return () => {
+      clearInterval(intervalId);
+    };
+  }, []);
+
+  const addNotification = useCallback(
+    (notification: NotificationItem, timeout = 5000) => {
```
Comment (soulBit) [inline apps/frontend/src/contexts/NotificationContext.tsx]: Move the 5 second default timeout to a constant with name `DEFAULT_TIMEOUT_SECONDS`

File: packages/ui/src/2_molecules/NotificationStack/NotificationStack.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+export default {
+  title: 'Molecule/NotificationStack',
+  component: NotificationStack,
+};
+
+const Template: Story<ComponentProps<typeof NotificationStack>> = args => {
+  const [items, setItems] = useState([...args.items]);
+  const onClose = (id: number | string) => {
+    console.log('onClose', id, items);
```
Comment (soulBit) [inline packages/ui/src/2_molecules/NotificationStack/NotificationStack.stories.tsx]:
```suggestion
```

File: packages/ui/src/2_molecules/NotificationStack/NotificationStack.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    setItems(items.filter(item => id !== item.id));
+  };
+  // fireTime and timeout
+  // now >fireTime + timeout;
```
Comment (soulBit) [inline packages/ui/src/2_molecules/NotificationStack/NotificationStack.stories.tsx]:
```suggestion
```

**Review by soulBit — CHANGES_REQUESTED** [review]

**Review by creed-victor — APPROVED** [review]

**Review by soulBit — APPROVED** [review]

**Review by LunaSov — APPROVED** [review]

**Pull request merged**